akka-cluster/src/main/scala/akka/cluster/Gossip.scala

```diff
@@ -156,59 +165,120 @@ private[cluster] final case class Gossip(
-    //      that have status DOWN, or
-    //   2. all unreachable members in the set have status DOWN or EXITING
-    // Else we can't continue to check for convergence
-    // When that is done we check that all members with a convergence
-    // status is in the seen table, i.e. has seen this version
-    val unreachable = reachabilityExcludingDownedObservers.allUnreachableOrTerminated.collect {
-      case node if (node != selfUniqueAddress && !exitingConfirmed(node)) ⇒ member(node)
+  def convergence(team: Team, selfUniqueAddress: UniqueAddress, exitingConfirmed: Set[UniqueAddress]): Boolean = {
+    // Find cluster members in the team that are unreachable from other members of the team
+    // excluding observations from members outside of the team, that have status DOWN or is passed in as confirmed exiting.
+    val unreachableInTeam = teamReachabilityExcludingDownedObservers(team).allUnreachableOrTerminated.collect {
+      case node if node != selfUniqueAddress && !exitingConfirmed(node) ⇒ member(node)
     }
-    unreachable.forall(m ⇒ Gossip.convergenceSkipUnreachableWithMemberStatus(m.status)) &&
-      !members.exists(m ⇒ Gossip.convergenceMemberStatus(m.status) &&
-        !(seenByNode(m.uniqueAddress) || exitingConfirmed(m.uniqueAddress)))
+
+    // If another member in the team that is UP or LEAVING and has not seen this gossip or is exiting
+    // convergence cannot be reached
+    def teamMemberHinderingConvergenceExists =
+      members.exists(member ⇒
+        member.team == team &&
+          Gossip.convergenceMemberStatus(member.status) &&
+          !(seenByNode(member.uniqueAddress) || exitingConfirmed(member.uniqueAddress))
+      )
+
+    // unreachables outside of the team or with status DOWN or EXITING does not affect convergence
+    def allUnreachablesCanBeIgnored =
+      unreachableInTeam.forall(unreachable ⇒ Gossip.convergenceSkipUnreachableWithMemberStatus(unreachable.status))
+
+    allUnreachablesCanBeIgnored && !teamMemberHinderingConvergenceExists
   }
 
   lazy val reachabilityExcludingDownedObservers: Reachability = {
     val downed = members.collect { case m if m.status == Down ⇒ m }
     overview.reachability.removeObservers(downed.map(_.uniqueAddress))
   }
 
-  def isLeader(node: UniqueAddress, selfUniqueAddress: UniqueAddress): Boolean =
-    leader(selfUniqueAddress).contains(node)
+  /**
+   * @return Reachability excluding observations from nodes outside of the team, but including observed unreachable
+   *         nodes outside of the team
+   */
+  def teamReachability(team: Team): Reachability =
+    overview.reachability.removeObservers(members.collect { case m if m.team != team ⇒ m.uniqueAddress })
+
+  /**
+   * @return reachability for team nodes, with observations from outside the team or from downed nodes filtered out
+   */
+  def teamReachabilityExcludingDownedObservers(team: Team): Reachability = {
```

Comment:
I think this is called many times, and the reason `reachabilityExcludingDownedObservers` is a `lazy val` is to avoid performing same filtering many times. The problem is that the `selfTeam` is passed in. Hmm, we can create a separate issue for looking at if we can refactor/optimize this.